libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -124,6 +124,16 @@ void ArrowAdapter::release_array(struct ArrowArray* array) {
         // If the ArrowBuffer.buffer_ shared_ptr is the last reference to the
         // underlying ColumnBuffer, the ColumnBuffer will be deleted.
         delete arrow_buffer;
+    } else {
+        // ArrowArray that allocate buffers directly to ``array->buffers`` will
+        // leak memory. We detect these arrays when the ``array->private_data``
+        // is null
+        for (int64_t i = 0; i < array->n_buffers; ++i) {
```

Comment:
this is probably a naive question, but why would we ever allocate directly into `buffers` without setting up the `private_data` structure?   (or just using `ArrowArrayInitFromType` to do it for us?)